### DISCUSSION TITLE
MinGW support

### DISCUSSION THREAD

**Discussion by forestf255** [eclipse-iceoryx/iceoryx#2013] [Q&A] [answered]
Hi all,

I'm interested in compiling iceoryx with MinGW. I've actually been able to do so locally off the v2.0.3 branch, the main lift was prepending `iox_` to definitions conflicting with MinGW's `pthread.h` in Window's [pthread.hpp file](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_platform/win/include/iceoryx_platform/pthread.hpp), i.e. `pthread_mutexattr_destroy` -> `iox_pthread_mutexattr_destroy`. This prefixing of pthread functions/structs will need also need to be updated for all other platforms.

What are the requirements for getting something like this upstreamed, particularly regarding testing?

Also has adding MinGW support previously been considered at all?

**Comment by elBoberido** [comment, marked as answer]
@forestf255 currently the platform is implicitly tested by the unit tests for `iceoryx_hoofs`. For upstreaming theses changes you need to
- sign the [ECA](https://www.eclipse.org/legal/ECA.php) (this is checked by the CI)
- create an issue (a few lines describing the problem are sufficient)
- use that issue number for commit messages, i.e. `iox-#42 Add 'iox' prefix to 'pthread' functions`
- update the release notes (just a one-liner with a link to the issue)
- create a PR

AFAIK we did not thought about adding MinGW. It might be a good idea to also add a CI target. Since we already have quite a lot of targets for the PRs it might be good to add this only to nightly builds. If it turns out to break easily, we can still promote it to the PR builds.

Looking forward to your contribution

**Reply by forestf255** [reply]
Thanks for the quick response!
